tools/js-tools/git-hooks/pre-push-hook.js

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
+
+/* eslint-disable no-console, no-process-exit */
+const { spawnSync } = require( 'child_process' );
+const chalk = require( 'chalk' );
+
+// Check if any touched files need a changelog file
+console.log( chalk.green( 'Checking if changelog files are needed. Just a sec...' ) );
+const needChangelog = checkChangelogFiles();
+
+// If a changelog file is needed, quit the push.
+if ( needChangelog.status === 1 ) {
+	process.exitCode = 1;
+}
+
+if ( needChangelog.status === 0 ) {
+	console.log( chalk.green( 'Changelog check passed.' ) );
+}
```

Comment:
What would you think of putting all this logic inside the `checkChangelogFiles()` function? That pattern seems clearer when I look at the checks in pre-commit-hooks.js.

Reply:
Works for me! Done in 5453055